### ISSUE TITLE
Reduce default autosave timeout

### ISSUE THREAD

**Issue by JasonWeill** [jupyterlab/jupyterlab#16892]
### Problem

The autosave timeout, introduced in #5645, is 120 seconds by default:

https://github.com/jupyterlab/jupyterlab/blob/090fe7383422b59588f0337fe0b5ee0d74899204/packages/docmanager-extension/schema/plugin.json#L119-L124

Users today expect autosave to work more quickly than every 2 minutes. A connection can easily become lost within 2 minutes, raising the risk of data loss.

### Proposed Solution

Reduce the autosave timeout to something much shorter by default, such as 5 seconds. Update all defaults, to match the new default.

### Additional context

Prior JupyterLab issue: #5619.

**Comment by Maduflavins** [comment]
@JasonWeill can I take up this task?

**Comment by JasonWeill** [comment]
@Maduflavins You can open a pull request if you want to, but I'd like to discuss this with our other contributors at triage next week before we merge any code in to address this.

**Comment by Maduflavins** [comment]
okay thank you

**Comment by Maduflavins** [comment]
Hi @JasonWeill I have opened a PR at https://github.com/jupyterlab/jupyterlab/pull/16893 you can have a look 
Thank you.

**Comment by Maduflavins** [comment]
@JasonWeill, I tried including the labels, but it looks like I do not have permission to do so. Can someone help me includeciated with this issue in the PR? 
Thank you.

**Comment by Maduflavins** [comment]
@JasonWeill  the regression ui test failed does this have anything to do with the changes i made. i do not see a link myself though and is there something i can do to resolve it meanwhile i saw your comment on the PR do you want me to make that change in this same PR?

**Comment by Maduflavins** [comment]
@JasonWeill did you get the chance to look at what @echarles said about the PR, I agree with his concern also I will be making changes to raise the default to 45secs against the 60secs he suggested

**Comment by JasonWeill** [comment]
@Maduflavins Thanks for opening this pull request! We plan to review this and other newly-filed issues on Tuesday at 09:00 PDT (16:00 UTC) as usual. You can find information about joining this triage meeting on the [Jupyter community calendar](https://discourse.jupyter.org/t/jupyter-community-calendar/2485), if you're interested and available.

**Comment by JasonWeill** [comment]
Per comments on #16893, including from @echarles , we need to determine what the impact is of saving a file more frequently than 2 minutes (120 seconds) after the most recent edit. Would decreasing the autosave timeout from 120s to 5s, for example, impose excessive load on the server?

**Comment by Maduflavins** [comment]
@JasonWeill whats is the decision on this issue do you think it’s wise to use 30secs instead of the 5 secs. Sorry I missed the triage meeting.

**Comment by JasonWeill** [comment]
@Maduflavins Based on our conversation last week, it seems like we need to inform any decision with data. Have you measured load on a server when there are multiple users editing a document with a 120-second autosave timeout, versus a 5-second or a 30-second timeout?

**Comment by Maduflavins** [comment]
@JasonWeill no, i have not. I will have to see if I can  see a use case to monitor.

**Comment by JasonWeill** [comment]
We may want to hold off on fixing this until #17017 is addressed; currently, notebooks in the hundreds of MB can take a long time to save, so a generally shorter autosave timeout might be bad for users with very large files.

**Comment by BaukJ** [comment]
Can across this issue and I'd just like to loop in this bug to make sure it's taken into consideration in discussions: https://github.com/jupyter/notebook/issues/7595

Decreasing the autosave interval may make that bug more prevalent.
(Although if it wasn't for that, I think I'd prefer a smaller autosave interval)
